test/connection/cross_site_forgery_test.rb

```diff
@@ -0,0 +1,55 @@
+require 'test_helper'
+require 'stubs/test_server'
+
+class ActionCable::Connection::CrossSiteForgeryTest < ActiveSupport::TestCase
```

Comment:
The Origin header is meant to be treated, and compared, as an opaque identifier. It will happen to parse as a valid URL, but it's a not a requirement we need to enforce. We just need to do straight equality comparison.